color_refiment_helper.py

```diff
@@ -8,51 +8,66 @@
 
 
 def debug(*args):
+    """
+    Prints debug statements when DEBUG is set to `True`
+
+    :param args: argument to be printed
+    :return:
+    """
     if DEBUG:
         print(*args)
 
 
-def compare(s: "List", t: "List"):
+def compare(s: "List", t: "List") -> bool:
+    """
+    Compares if two lists are equal
+
+    :param s: List to compare
+    :param t: List to compare
+    :return: Returns `True` if list have the same length and contain the same elements. Elements do not need to have the
+    same order. Returns `False` otherwise.
+    """
+    # TODO: change internal comparison since `Counter` may not be used
     return Counter(s) == Counter(t)
 
 
-def create_partition(old_coloring: "Coloring", vertex1: "Vertex", vertex2: "Vertex") -> "Coloring":
+def create_partition(coloring: Coloring, vertex1: Vertex, vertex2: Vertex) -> Coloring:
     """
-    Returns a new coloring where both vertices are in a new partition and removed from the one they belonged to.
+    Returns a new coloring where both vertices have the same new color and are removed from the one they belonged to
 
-    :param old_coloring: current coloring
-    :param vertex1: vertex to be in the separate partition
-    :param vertex2: vertex to be in the separate partition
-    :return: a new coloring with vertex1 and vertex2 as a seperate partition
+    :param coloring: current coloring
+    :param vertex1: vertex to be in the separate color
+    :param vertex2: vertex to be in the separate color
+    :return: a new coloring with vertex1 and vertex2 as a separate color
     """
-    new_coloring = old_coloring.copy()
+    new_coloring = coloring.copy()
     new_color = new_coloring.next_color()
     new_coloring.recolor(vertex1, new_color)
     new_coloring.recolor(vertex2, new_color)
     return new_coloring
 
 
-def has_same_color_neighbours(u: "Vertex", v: "Vertex", coloring: "Coloring") -> bool:
+def has_same_color_neighbours(u: Vertex, v: Vertex, coloring: Coloring) -> bool:
```

Comment:
Waarom mag je hier `Vertex` gebruiken (ipv `"Vertex"`) maar is het eerder wel `"List"` ipv `List`?

Reply:
Vraag dat aan Python...

Reply:
Overal mag zonder accentjes, behalve op een plek waar bijv. `Vertex` nog niet een bekende naam is, bijv. in de `Vertex` klasse zelf (daar wordt de klasse namelijk nog gedefinieerd i.p.v. dat het een bekende naam is). Zo kun je ook recursieve type aliases maken, bijv.:

```python
from typing import Any, Union

BinaryTree = Union['Leaf', 'Node']  # Here, Leaf and Node are referenced before being declared!
Leaf: BinaryTree = Any
Node: BinaryTree = (BinaryTree, Any, BinaryTree)  # Tuple of left tree, node value and right tree, recursive!

if __name__ == '__main__':
    leaf_0: Leaf = 0
    leaf_1: Leaf = 1
    leaf_2: Leaf = 2
    node_3: Node = (leaf_0, 3, None)
    node_4: Node = (leaf_1, 4, leaf_2)
    node_5: Node = (node_3, 5, node_4)

    # Now, we have the following binary tree:
    #        5
    #       / \
    #      /   \
    #     /     \
    #    /       \
    #   3         4
    #  / \       / \
    # 0   None  1   2
```

Reply:
Overigens is mijn voorbeeld totaal niet 'Pythonesque': al die 'typen' zijn voor de sier en hartstikke dynamisch. Haal ze allemaal weg en het werkt identiek, zo niet sneller. Echter, je IDE kan misschien iets meer type hints geven (warnings, nooit errors) wat het ontwikkelen misschien wat helpt. Zeker voor ons met ervaring in Java.